QMigrate/signal/local_mag/magnitude.py

```diff
@@ -0,0 +1,997 @@
+            each station in the station file, and local magnitude estimates
+            calculated from them (output by calculate_magnitudes()). Note that
+            the amplitude observations are raw, but the ML estimates derived
+            from them include station corrections, if provided.
+            Columns = ["epi_dist", "z_dist", "P_amp", "P_freq", "P_time",
+                       "S_amp", "S_freq", "S_time", "Noise_amp", "is_picked",
+                       "ML", "ML_Err"]
+
+        Returns
+        -------
+        used_mags : `pandas.DataFrame`
+            As input, but only including individual amplitude measurements /
+            local magnitude estimates that meet the filters specified by the
+            user. Now with additional columns:
+            Noise_Filter : bool
+                Whether this observation meets the noise filter.
+            Trace_Filter : bool
+                Whether this observation matches the trace filter.
+            Station_Filter : bool
+                Whether this observation is not excluded by the station filter.
+            Dist_Filter : bool
+                Whether this observation meets the distance filter.
+            Dist : bool
+                The (epicentral or hypocentral) distance between the station
+                and event.
+            Used : bool (== True)
+                Whether the observation meets all filter requirements.
+        all_mags : `pandas.DataFrame`
+            As for used_mags, but containing all observations from the input
+            magnitudes DataFrame, other than those which feature null values
+            for the signal or noise amplitude.
+
+        """
+
+        # Remove nan amplitude values
+        magnitudes.dropna(subset=[self.amp_feature, "Noise_amp"], inplace=True)
+        # magnitudes = magnitudes.loc[~np.isnan(amps), :]
+        # magnitudes = magnitudes.[~(pd.isnull(magnitudes[self.amp_feature]))]
+
+        # Apply noise filter.
+        if self.noise_filter:
+            amps = magnitudes[self.amp_feature].values
+            noise_amps = magnitudes["Noise_amp"].values
+            magnitudes["Noise_Filter"] = False
+            with np.errstate(invalid="ignore"):
+                magnitudes.loc[(amps > noise_amps * self.noise_filter),
+                               "Noise_Filter"] = True
+
+        # Apply trace filter
+        if self.trace_filter:
+            magnitudes["Trace_Filter"] = False
+            magnitudes.loc[magnitudes.index.str.contains(self.trace_filter),
+                           "Trace_Filter"] = True
+
+        # Apply station filter
+        if self.station_filter:
+            magnitudes["Station_Filter"] = True
+            for stn in list(self.station_filter):
+                magnitudes.loc[magnitudes.index.str.contains(f".{stn}.",
+                                                             regex=False),
+                               "Station_Filter"] = False
```

Comment:
```suggestion
        if self.noise_filter is not None:
            amps = magnitudes[self.amp_feature].values
            noise_amps = magnitudes["Noise_amp"].values
            magnitudes["Noise_Filter"] = False
            with np.errstate(invalid="ignore"):
                magnitudes.loc[(amps > noise_amps * self.noise_filter),
                               "Noise_Filter"] = True

        # Apply trace filter
        if self.trace_filter:
            magnitudes["Trace_Filter"] = False
            magnitudes.loc[magnitudes.index.str.contains(self.trace_filter),
                           "Trace_Filter"] = True

        # Apply station filter
        if self.station_filter is not None:
            magnitudes["Station_Filter"] = True
            for stn in list(self.station_filter):
                magnitudes.loc[magnitudes.index.str.contains(f".{stn}.",
                                                             regex=False),
                               "Station_Filter"] = False
```

Reply:
Deliberate to apply this to `self.noise_filter` (should be a float) and `self.station_filter` (agreed) but not `self.trace_filter` ?

I'm not sure whether it's best to have an if for `self.noise_filter` or not. If it's strictly a float, then the if should really be to check if it's not 0.

Reply:
```suggestion
        if self.noise_filter != 0.:
            amps = magnitudes[self.amp_feature].values
            noise_amps = magnitudes["Noise_amp"].values
            magnitudes["Noise_Filter"] = False
            with np.errstate(invalid="ignore"):
                magnitudes.loc[(amps > noise_amps * self.noise_filter),
                               "Noise_Filter"] = True

        # Apply trace filter
        if self.trace_filter is not None:
            magnitudes["Trace_Filter"] = False
            magnitudes.loc[magnitudes.index.str.contains(self.trace_filter),
                           "Trace_Filter"] = True

        # Apply station filter
        if self.station_filter is not None:
            magnitudes["Station_Filter"] = True
            for stn in list(self.station_filter):
                magnitudes.loc[magnitudes.index.str.contains(f".{stn}.",
                                                             regex=False),
                               "Station_Filter"] = False
```